### PR TITLE
Put back video metadata

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I updated the `video-metadata-rs` crate: now, no more ffmpeg, just pure rust. The webm format isn't checked yet.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13094)

### CODE REVIEW

**Comment by KiChjang** [comment]
Does this require additional libraries to be installed on users' system?

**Comment by GuillaumeGomez** [comment]
No. That's the whole point and the great improvement.

**Comment by jdm** [comment]
cc @splav This will impact the plans for using mp4parse.

**Comment by jdm** [comment]
r? @Manishearth 
By random selection, since I don't intend to review code this week.

**Comment by GuillaumeGomez** [comment]
@jdm: I'll put @Ms2ger on it then.

r? @Ms2ger

**Comment by est31** [comment]
Related : #13024, #13024.

**Comment by splav** [comment]
First: The current metadata structure seems to be suitable only for video content. Are there any plans to support audio content as well? May we should make it more general to support both?
Second: mp4_parse essentially does not have rust API yet. So the crate uses own patched version that makes some structs public. I saw the PR for mp4_parse, but I don't think it's a good long-term solution and that it will be used.

**Comment by GuillaumeGomez** [comment]
I can make it more general (by providing a higher crate maybe? The name is video-metadata after all). And as for mp4parse, as long as it doesn't need strange dependencies, it's fine for the moment from my point of view.

**Comment by splav** [comment]
@GuillaumeGomez I was talking about Servo's structure. But sure, this can be changed later.

**Comment by splav** [comment]
As for video-metadata - may be it's ok, but ogg can contain audio as well and checking it twice in two crated seems a bit strange.

**Comment by GuillaumeGomez** [comment]
@splav: For the moment, the thing is to target video to get their size. But if there are pure crates to get audio metadata, I can update video-metadata to add them (I still wonder if I should rename it...).

**Comment by est31** [comment]
My `ogg_metadata` crate which `video-metadata-rs` uses supports audio metadata parsing as well as video metadata parsing: it returns the duration and the sample frequency (at least for vorbis, opus has no fixed sampling frequency).

**Comment by splav** [comment]
@est31, @GuillaumeGomez my point that it should be one metadata wrapper crate for servo for both audio and video containers. The other option is to have one crate for audio and one for video leads to parsing ogg container twice: in both video and audio crates. It seems a bit strange for me.

And IMO the crate rename is the thing that should be done the earlier the better, in case the crate will support both audio and video metadata parsing.

**Comment by GuillaumeGomez** [comment]
I created the new crate (which handles both video and audio metadata) and updated this PR.

File: components/script/dom/htmlmediaelement.rs [hunk 1]
```diff
@@ -160,12 +161,24 @@ impl HTMLMediaElementContext {
...
-        // TODO: Properly implement once we have figured out the build and
-        // licensing ffmpeg issues.
-        elem.change_ready_state(HAVE_METADATA);
-        self.have_metadata = true;
+        match audio_video_metadata::get_format_from_slice(&self.data) {
+            Ok(audio_video_metadata::Metadata::Video(meta)) => {
+                let dur = meta.audio.duration.unwrap_or(::std::time::Duration::new(0, 0));
+                *elem.video.borrow_mut() = Some(VideoMedia {
+                    format: format!("{:?}", meta.format),
+                    duration: Duration::seconds(dur.as_secs() as i64) +
```
Comment (est31) [inline components/script/dom/htmlmediaelement.rs]: Maybe use Duration::from_std() here?

Comment (splav) [inline components/script/dom/htmlmediaelement.rs]: And why not just clone?

Comment (est31) [inline components/script/dom/htmlmediaelement.rs]: @splav dur is of type `std::time::Duration`, while the VideoMedia struct requires `time::Duration` from the time crate.

Comment (splav) [inline components/script/dom/htmlmediaelement.rs]: Oh, yep.

**Comment by nox** [comment]
Think you could publish audio_video_metadata? Would be great to avoid yet another Git dependency.

**Comment by GuillaumeGomez** [comment]
@nox: No, I have mixed origin for this crate (because mp4rust didn't merge my PR yet) so I can't publish it for the moment.

**Comment by nox** [comment]
https://github.com/mozilla/mp4parse-rust/pull/21

**Comment by GuillaumeGomez** [comment]
So `audio-video-metadata` crate is now on crates.io. It now also supports MP3.

**Comment by est31** [comment]
@nox that PR got merged, is this still blocked on external?

**Comment by nox** [comment]
Yes, mp4parse 0.5 wasn't released.

**Comment by GuillaumeGomez** [comment]
The mp4parse has been released and I updated audio-video-metadata crate.

**Comment by est31** [comment]
@nox I think now labels can be updated, no?

**Comment by nox** [comment]
The Cargo.lock files are outdated.

**Comment by GuillaumeGomez** [comment]
Updated.

File: servo-tidy.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ skip-check-licenses = false
 
 [ignore]
 # Ignored packages with duplicated versions
-packages = ["lazy_static"]
+packages = ["lazy_static", "bitflags"]
```
Comment (Ms2ger) [inline servo-tidy.toml]: Why?

**Comment by nox** [comment]
The code looks ok but I don't really want to r+ something that introduces a bunch of timeouts. Cc @Ms2ger @jdm.

**Comment by nox** [comment]
Not only this PR introduces timeouts, but it doesn't seem to make any new test pass.

File: servo-tidy.toml (hunk 2 above)
Comment (GuillaumeGomez) [inline servo-tidy.toml]: It was because `mp4parse-rust` depended on rust-cheddar. It's not the case anymore so I remove it.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by jdm** [comment]
The timeouts are fine, since the reflect the actual state of the world when it comes to audio elements. The current HTMLMediaElement code just lies about understanding audio metadata.

**Comment by KiChjang** [comment]
@bors-servo r+
* Little PIP-y boy is annoying

**Pull request merged**